cmd/parse/blocks/blocks.go

```diff
@@ -44,8 +44,18 @@ will be replaced with the data downloaded from the node.
 			end, _ := cmd.Flags().GetInt64(flagEnd)
 			force, _ := cmd.Flags().GetBool(flagForce)
 
+			lastBlockHeight, err := parseCtx.Database.GetLastBlockHeight()
+			if err != nil {
+				return err
+			}
+
 			// Get the start height, default to the config's height; use flagStart if set
 			startHeight := config.Cfg.Parser.StartHeight
+
+			if lastBlockHeight > startHeight {
+				startHeight = lastBlockHeight
+			}
+
```

Comment:
I think a better way to solve this is to use the following instead:
```go
startHeight := max.MaxInt64(config.Cfg.Parser.StartHeight, lastDbBlockHeight)
```

(Note: `lastBlockHeight` should be renamed to `lastDbBlockHeight` to make it easier to understand)